hash_practice/exercises.py

```diff
@@ -1,20 +1,52 @@
 
-def grouped_anagrams(strings):
+def grouped_anagrams(d1):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
         Time Complexity: ?
         Space Complexity: ?
     """
-    pass
+    # empty dictionary for anagrams together 
+    dict = {} 
+
+    # traversal 
+    for val in d1: 
+        # sorts list
+        key = ''.join(sorted(val)) 
+
+        if key in dict.keys(): 
+            dict[key].append(val) 
+        else: 
+            dict[key] = [] 
+            dict[key].append(val) 
+
+    # traverse dictionary and join keys together 
+    result = []
+    for key,value in dict.items(): 
+        # result = result + ' '.join(value) + ' '
+        result.append(value)
+    return result 
 
-def top_k_frequent_elements(nums, k):
+def top_k_frequent_elements(numbers, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: o(n)
+        Space Complexity: o(n)
     """

```

Comment:
👍 However because you're finding max of a list of n elements k times, this is O(nk) for time complexity.